commands/deploy_command.go

```diff
@@ -64,15 +64,17 @@ var resourcesList listFlag
 
 // DeployCommand is a command for deploying an MTA archive
 type DeployCommand struct {
-	BaseCommand
-	defineCommandFlags   CommandFlagsDefiner
+	*BaseCommand
 	setProcessParameters ProcessParametersSetter
 	processTypeProvider  ProcessTypeProvider
 }
 
 // NewDeployCommand creates a new deploy command.
 func NewDeployCommand() *DeployCommand {
-	return &DeployCommand{BaseCommand{}, deployCommandFlagsDefiner(), deployProcessParametersSetter(), &deployCommandProcessTypeProvider{}}
+	baseCmd := &BaseCommand{flagsParser: newDeployCommandLineArgumentsParser(), flagsValidator: deployCommandFlagsValidator{}}
+	deployCmd := &DeployCommand{baseCmd, deployProcessParametersSetter(), &deployCommandProcessTypeProvider{}}
+	baseCmd.Command = deployCmd
```

Comment:
These seem like they can be missed. Is there a way to be set in something like a construction function?
NewDeployCommand -> inside this function baseCmd.Command will be initialized with something like "this"